sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -126,7 +140,18 @@ public async ValueTask<string> CreateRequestFailedMessageAsync(Response response
 
         public string CreateRequestFailedMessageWithContent(Response response, string? message, string? content, string? errorCode, IDictionary<string, string>? additionalInfo)
```

Comment:
This method looks very similar to CreateRequestFailedMessageWithContent. 

If you were to add `IDictionary<string, string>? additionalInfo` parameter to `CreateRequestFailedMessageWithContent` you can avoid splitting formatting into many submethods. ResponseClassifier would pass null to additionalInfo

Reply:
It's a fair point.  I liked that this was starting to factor out what's part of the HTTP message vs. what's a customization added in a service library.  I think that's hard to tease out the way the class is right now, and would be useful in the future.  But I see your point about code churn as well.